simplq/src/components/pages/Admin/QueueSettings.jsx

```diff
@@ -17,11 +17,14 @@ const MAX_SIZE = 100000;
 export default ({ queueId }) => {
   const [isModalOpen, setIsModalOpen] = useState(false);
   const maxQueueSize = useSelector(selectMaxQueueCapacity);
+  const isSelfJoinAllowed = useSelector(selectIsSelfJoinAllowed);
   const [size, setSize] = React.useState(maxQueueSize);
+  const [selfJoin, setSelfJoin] = React.useState(false);
 
   useEffect(() => {
     setSize(maxQueueSize);
-  }, [maxQueueSize]);
+    setSelfJoin(!!isSelfJoinAllowed);
+  }, [maxQueueSize, isSelfJoinAllowed]);
```

Comment:
I'm curious about the !!. Is this meant to default to false if the value is undefined or is it for something else?

Reply:
Exactly. Default to false if value is undefined